docs/src/diagnostic_edmf_equations.md

```diff
@@ -0,0 +1,76 @@
+# Diagnostic EDMF equations
+
+This describes the diagnostic EDMF scheme equations and its discretizations. Where possible, we use a coordinate invariant form: the ClimaCore operators generally handle the conversions between bases internally.
+
+
+## Grid-scale variables
+
+* ``\rho``: _density_ in kg/m³, discretized at cell centers.
+* ``u^3``: the contravariant 3 component of velocity, discretized at cell faces.
+* ``h_{\mathrm{tot}}``: total enthalpy in J/kg, discretized at cell centers.
+* ``q_t``: total specific humidity in kg/kg, discretized at cell centers.
+* ``\Phi = g z``: geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level, discretized at cell centers.
+* ``\nabla^3 \Phi``: the contravariant 3 component of the gradient of geopotential, reconstructed at cell centers.
+* ``p``: air pressure, reconstructed at cell centers.
+
+## Subgrid-scale variables
+* ``\hat{\rho}^j``: _effective density_ in kg/m³. Superscript ``j`` represents the sub-domain. ``\hat{\rho}^j = \rho^j a^j`` where ``\rho^j`` is the sub-domain density and ``a^j`` is the sub-domain area fraction. This is discretized at cell centers.
+* ``\rho^j``: _density_ in kg/m³, derived from the thermodynamic state, reconstructed at cell centers.
+* ``u^{j,3}``: The contravariant 3 component of velocity, discretized at cell faces.
+* ``h_{\mathrm{tot}}^j``: Total enthalpy, discretized at cell centers.
+* ``q_t^j``: Total specific humidity of the sub-domain j, discretized at cell centers.
+* ``u^{0,3}``: The contravariant 3 component of velocity of the environment, obtained as residuals:
+  ```math
+  \rho u^{0, 3} = \rho u^3 -  \sum_{j\ne 0} \hat\rho^j u^{j, 3}ß
+  ```
+
+## Equations and discretizations
+
+### Mass
+
+```math
+\frac{1}{J} \frac{\partial}{\partial \xi^3} \bigl( \hat\rho^j J u^{j, 3} \bigr)
+= (E^{j0} - \Delta^{j0}) \hat\rho^j
+```
+
+This is descritized using the following
+```math
+\frac{1}{J[i-1]} \left( J[i-\frac{1}{2}] \hat\rho^j[i] u^{(j), 3}[i-\frac{1}{2}] -J[i-\frac{3}{2}] \hat\rho^j[i-1] u^{j, 3}[i-\frac{3}{2}] \right)
```

Comment:
Wouldn't it be easier to write (here and in the other equations) by shifting indices +1, i.e., 
```suggestion
\frac{1}{J[i]} ...
```

Reply:
That works too (and is probably easier). I was trying to follow the code, where we solve e.g. \rho[i] and u[i-1/2] using \rho[i-1] and u[i-3/2] at index i in the loop.